GFG.java

```diff
@@ -1,59 +1,87 @@
-import java.util.LinkedList; 
-import java.util.Queue; 
-public class GFG { 
-       
-    /* A binary tree node has key, pointer to  
-    left child and a pointer to right child */
-    static class Node { 
-        int key; 
-        Node left, right; 
-          
-        // constructor 
-        Node(int key){ 
-            this.key = key; 
-            left = null; 
-            right = null; 
-        } 
-    } 
-    static Node root; 
-    static Node temp = root; 
-      
-    /* Inorder traversal of a binary tree*/
-    static void inorder(Node temp) 
-    { 
-        if (temp == null) 
-            return; 
-       
-        inorder(temp.left); 
-        System.out.print(temp.key+" "); 
-        inorder(temp.right); 
-    } 
-       
-    /*function to insert element in binary tree */
-    static void insert(Node temp, int key) 
-    { 
-
-        // Do level order traversal until we find 
-        // an empty place and add the node.  
-    } 
-       
-    // Driver code 
-    public static void main(String args[]) 
-    { 
-        root = new Node(10); 
-        root.left = new Node(11); 
-        root.left.left = new Node(7); 
-        root.right = new Node(9); 
-        root.right.left = new Node(15); 
-        root.right.right = new Node(8); 
-       
-        System.out.print( "Inorder traversal before insertion:"); 
-        inorder(root); 
-       
-        int key = 12; 
-        insert(root, key); 
-       
-        System.out.print("\nInorder traversal after insertion:"); 
-        inorder(root); 
-    } 
+import java.util.LinkedList; 
+import java.util.Queue; 
+public class GFG { 
+       
+    /* A binary tree node has key, pointer to  
+    left child and a pointer to right child */
+    static class Node { 
+        int key; 
+        Node left, right; 
+          
+        // constructor 
+        Node(int key){ 
+            this.key = key; 
+            left = null; 
+            right = null; 
+        } 
+    } 
+    static Node root; 
+    static Node temp = root; 
+      
+    /* Inorder traversal of a binary tree*/
+    static void inorder(Node temp) 
+    { 
+        if (temp == null) 
+            return; 
+       
+        inorder(temp.left); 
+        System.out.print(temp.key+" "); 
+        inorder(temp.right); 
+    } 
+       
+    /*function to insert element in binary tree */
+    static void insert(Node temp, int key) 
+    { 
+
+        // Do level order traversal until we find 
+        // an empty place and add the node.  
+        if(temp == null){
+            temp = new Node(key);
+            root = temp;
```

Comment:
Can you explain reason to line 40 and 41? 
Is this actually required?

Reply:
I added with intention to create root node if the tree is empty. Now realized since the insert method doesn't return anything we can't create a root node that way.

Reply:
Yeah!! Exactly.
Good to see that you got that yourself.